modules/system/assets/js/snowboard/ajax/Request.js

```diff
@@ -0,0 +1,789 @@
+                            } else {
+                                this.processResponse(response);
+                            }
+                        },
+                    );
+                },
+                (error) => {
+                    this.responseError = error;
+                    this.processError(error);
+                },
+            ).finally(() => {
+                if (this.cancelled === true) {
+                    return;
+                }
+
+                if (this.options.complete && typeof this.options.complete === 'function') {
+                    this.options.complete(this.responseData, this);
+                }
+                this.snowboard.globalEvent('ajaxDone', this.responseData, this);
+
+                if (this.element) {
+                    const event = new Event('ajaxAlways');
+                    event.request = this;
+                    event.responseData = this.responseData;
+                    event.responseError = this.responseError;
+                    this.element.dispatchEvent(event);
+                }
+            });
+        }
+    }
+
+    /**
+     * Dependencies for this plugin.
+     *
+     * @returns {string[]}
+     */
+    dependencies() {
+        return ['jsonParser', 'sanitizer'];
+    }
+
+    /**
+     * Validates the element and handler given in the request.
+     */
+    checkRequest() {
+        if (this.element !== undefined && this.element instanceof Element === false) {
+            throw new Error('The element provided must be an Element instance');
+        }
+
+        if (this.handler === undefined) {
+            throw new Error('The AJAX handler name is not specified.');
+        }
+
+        if (!this.handler.match(/^(?:\w+:{2})?on*/)) {
+            throw new Error('Invalid AJAX handler name. The correct handler name format is: "onEvent".');
+        }
+    }
+
+    /**
+     * Run client-side validation on the form, if available.
+     *
+     * @returns {boolean}
+     */
+    doClientValidation() {
+        if (this.options.browserValidate === true && this.form) {
+            if (this.form.checkValidity() === false) {
+                this.form.reportValidity();
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Executes the AJAX query.
+     *
+     * Returns a Promise object for when the AJAX request is completed.
+     *
+     * @returns {Promise}
+     */
+    doAjax() {
+        // Allow plugins to cancel the AJAX request before sending
+        if (this.snowboard.globalEvent('ajaxBeforeSend', this) === false) {
+            return Promise.resolve({
+                cancelled: true,
+            });
+        }
+
+        const ajaxPromise = new Promise((resolve, reject) => {
+            fetch(this.url, {
+                method: 'POST',
+                headers: this.headers,
+                body: this.data,
+                redirect: 'follow',
+                mode: 'same-origin',
```

Comment:
This might be something that we explore at a later date, but is it possible to make AJAX requests to another domain with the current code? I know I've done that with the old framework once before (required setting up the CORS plugin but otherwise worked) in order to embed the framework and a simple use of it on an external site that wasn't powered by Winter in order to still be able to use some of Winter's power.

Reply:
I did think about this the other day. I think I would be happy with allowing remote calls using Request, but not necessarily accepting partial changes from them or allowing them to directly impact the app - so you could use them to retrieve data from, say, a JSON API but then would need to handle the data and page manipulations yourself.

Reply:
I'm not sure why we would care about restricting based on that, presumably if the developer is making calls to an external service they accept the risk. Also of note my example was loading the js from example.com on notexample.com, example.com being the winter instance that was both the host of the js and the target url, while notexample.com was just pulling in the framework and sending Ajax requests to example.com in order to have the power of the Ajax framework in a WordPress frontend

Reply:
Alright, fair enough.

The URL was modifiable through the options, but the `fetch()` config was not. I've just pushed an update to allow the fetch config to be modified via the `options`, or through a global event.